tests/fsharp/nunitConf.fs

```diff
@@ -0,0 +1,379 @@
+
+type FSharpSuiteTestCaseData =
+    inherit TestCaseData
+
+    new (dir: string, [<ParamArray>] arguments: Object array) as this = 
+        { inherit TestCaseData(arguments) }
+        then
+            this.Properties |> FSharpTestSuite.setProps dir
+            arguments
+            |> Array.choose (fun a -> match a with :? Permutation as p -> Some p | _ -> None)
+            |> Array.iter (fun p -> this.SetCategory(sprintf "%A" p) |> ignore)
+
+[<AttributeUsage(AttributeTargets.Method, AllowMultiple = true, Inherited = false)>]
+type FSharpSuiteTestCaseAttribute =
+    inherit TestCaseAttribute
+
+    new (dir: string, [<ParamArray>] arguments: Object array) as this = 
+        { inherit TestCaseAttribute(arguments) }
+        then
+            this.Properties |> FSharpTestSuite.setProps dir
+
+
+type FSharpSuitePermutationsAttribute(dir: string) =
+    inherit NUnitAttribute()
+
+    let _builder = NUnit.Framework.Internal.Builders.NUnitTestCaseBuilder()
+    interface NUnit.Framework.Interfaces.ITestBuilder with
+        member x.BuildFrom(methodInfo, suite) =
+            let allPermutations = 
+                [ FSI_FILE; FSI_STDIN; FSI_STDIN_OPT; FSI_STDIN_GUI;
+                  FSC_BASIC; FSC_HW; FSC_O3;
+                  GENERATED_SIGNATURE; EMPTY_SIGNATURE; EMPTY_SIGNATURE_OPT; 
+                  FSC_OPT_MINUS_DEBUG; FSC_OPT_PLUS_DEBUG; 
+                  FRENCH; SPANISH;
+                  AS_DLL; 
+                  WRAPPER_NAMESPACE; WRAPPER_NAMESPACE_OPT 
+                ]
+                |> List.map (fun p -> (new FSharpSuiteTestCaseData (dir, p)))
+
+            allPermutations
+            |> List.map (fun tc -> _builder.BuildTestMethod(methodInfo, suite, tc))
+            |> Seq.ofList
+
+module FileGuard =
+    let private remove path = if File.Exists(path) then Commands.rm (Path.GetTempPath()) path
+
+    [<AllowNullLiteral>]
+    type T (path: string) =
+        member x.Path = path
+        interface IDisposable with
+            member x.Dispose () = remove path
+
+    let create path =
+        if not (Path.IsPathRooted(path)) then failwithf "path '%s' must be absolute" path
+        remove path
+        new T(path)
+    
+    let exists (guard: T) = guard.Path |> File.Exists
+        
+
+let checkGuardExists guard = processor {
+    if not <| (guard |> FileGuard.exists)
+    then return! genericError (sprintf "exit code 0 but %s file doesn't exists" (guard.Path |> Path.GetFileName))
+    }
+
+
+let check (f: Attempt<_,_>) =
+    f |> Attempt.Run |> checkTestResult
+
+
+type RedirectInfo = 
+    { Output : RedirectTo
+      Input : RedirectFrom option }
+
+and RedirectTo = 
+    | Inherit
+    | Output of RedirectToType
+    | OutputAndError of RedirectToType
+    | Error of RedirectToType
+
+and RedirectToType = 
+    | Overwrite of FilePath
+    | Append of FilePath
+
+and RedirectFrom = 
+    | RedirectInput of FilePath
+
+
+module Command =
+
+    let logExec dir path args redirect =
```

Comment:
And add a single line /// comment for each to type definition too please. thanks!